### PR TITLE
[5.4] Add a preg_fetch() helper

### PR DESCRIPTION
**Pull request by caneco**
Add a `preg_fetch()` helper to get on the fly the `$matches` from a normal `preg_match()`.

```PHP
$matches = preg_fetch("/^(.+)@/", "foo@bar.com");

// returns: `["foo"]`
```

### CODE REVIEW

**Comment by caneco** [comment]
Newbie question around StyleCI … shouldn't be there a new line?

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -614,6 +614,23 @@ function object_get($object, $key, $default = null)
...
+     *
+     * @param  string  $pattern
+     * @param  string  $subject
+     * @param  int     $flags
+     * @return array
+     */
+    function preg_fetch($pattern, $subject, $flags = 0)
+    {
+        $result = preg_match($pattern, $subject, $matches, $flags);
+
```
Comment (Lloople) [inline src/Illuminate/Support/helpers.php]: I think StyleCI is telling you that you have spaces there and it wants to be a new empty line, but not sure

**Review by Lloople — CHANGES_REQUESTED** [review]

**Comment by lucasmichot** [comment]
tests ?

File: src/Illuminate/Support/helpers.php [hunk 2]
```diff
@@ -614,6 +614,23 @@ function object_get($object, $key, $default = null)
...
+     * Perform a regular expression match and returns the matches.
+     *
+     * @param  string  $pattern
+     * @param  string  $subject
+     * @param  int     $flags
+     * @return array
+     */
+    function preg_fetch($pattern, $subject, $flags = 0)
+    {
+        $result = preg_match($pattern, $subject, $matches, $flags);
```
Comment (lucasmichot) [inline src/Illuminate/Support/helpers.php]: No need to set `$result`

**Comment by taylorotwell** [comment]
I don't typically add helpers the framework itself doesn't use.

**Pull request closed without merging**